src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -305,6 +306,42 @@ public function orWhere($column, $operator = null, $value = null)
         return $this->where($column, $operator, $value, 'or');
     }
 
+    /**
+     * Add a "BelongsTo" relationship where clause to the query.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $related
+     * @param  string  $relationship
+     * @param  string  $boolean
+     * @return $this
+     *
+     * @throws \Exception
+     */
+    public function whereBelongsTo($related, $relationshipName = null, $boolean = 'and')
```

Comment:
one more thing: isn't this better suited for the trait `Illuminate\Database\Eloquent\Concerns\QueriesRelationships`? It appears that most of the relation-related functions appear in that trait instead of the Builder class itself.

(although there are some that are still in the builder like `with()`/`without()`/`withOnly()`/`eagerLoadRelation()`/`getRelation()` which might be considered "querying relationships")

Reply:
good point, moved it